src/widgets/ProtectedRoute/InactivityTracker.tsx

```diff
@@ -1,50 +1,34 @@
-import { PropsWithChildren, useCallback, useEffect } from 'react';
+import { PropsWithChildren, useEffect } from 'react';
 
+import { useIdleTimer, events } from '~/features/IdleTimer';
 import { useLogout } from '~/features/Logout';
-import useTimer from '~/shared/utils/useTimer';
 
 export type InactivityTrackerProps = PropsWithChildren<unknown>;
 
-const events = ['load', 'click', 'scroll', 'keypress', 'mousemove'];
-
-const ONE_SEC = 1000;
-const ONE_MIN = 60 * ONE_SEC;
-const LOGOUT_TIME_LIMIT = 15 * ONE_MIN; // 15 min
-
 export const InactivityTracker = ({ children }: InactivityTrackerProps) => {
   const { logout } = useLogout();
-  const { resetTimer, setTimer } = useTimer();
-
-  const onLogoutTimerExpire = useCallback(() => {
-    // Listener clean up. Removes the existing event listener from the window
-    Object.values(events).forEach((item) => {
-      window.removeEventListener(item, resetTimer);
-    });
 
-    // logs out user
-    logout();
-  }, [logout, resetTimer]);
-
-  const onActivityEventHandler = useCallback(() => {
-    resetTimer();
-    setTimer({
-      time: LOGOUT_TIME_LIMIT,
-      onComplete: onLogoutTimerExpire,
-      timerName: 'InactivityTracker',
-    });
-  }, [resetTimer, setTimer, onLogoutTimerExpire]);
+  const IdleTimer = useIdleTimer({
+    time: {
+      hours: 0,
+      minutes: 15,
+    },
+    onFinish: logout,
+  });
 
   useEffect(() => {
-    Object.values(events).forEach((item) => {
-      window.addEventListener(item, onActivityEventHandler);
+    const listener = () => IdleTimer.start('InactivityTracker');
+
+    events.forEach((item) => {
+      window.addEventListener(item, listener);
     });
 
     return () => {
-      Object.values(events).forEach((item) => {
-        window.removeEventListener(item, onActivityEventHandler);
+      events.forEach((item) => {
+        window.removeEventListener(item, listener);
       });
     };
-  }, [onActivityEventHandler]);
+  }, [IdleTimer]);
```

Comment:
Please double-check to ensure everything is working the way it should. `IdleTimer` is recreated on every render by `useIdleTimer`. Hence, the effect and its clean function run every render.

Reply:
Done